### PR TITLE
Fixes and improvements

### PR DESCRIPTION
**Pull request by martingrossmann**
# Description

This is a summery of different changes:
* Fix NPE in case of empty method context
* Updated to JDK11
* Updated Selenoid tests to Testerra 2

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/main/java/eu/tsystems/mms/tic/testerra/plugins/selenoid/webdriver/SelenoidCapabilityProvider.java [hunk 1]
```diff
@@ -87,7 +87,10 @@ private Capabilities provide(DesktopWebDriverRequest request) {
         final String runConfigName = contextController.getExecutionContext().getRunConfig().RUNCFG;
 
         // Try to find out the current testmethod to add the name to the Selenoid caps
-        String methodName = contextController.getCurrentMethodContext().get().getName();
+
+        String methodName = contextController.getCurrentMethodContext().isPresent()
```
Comment (Zsar) [inline src/main/java/eu/tsystems/mms/tic/testerra/plugins/selenoid/webdriver/SelenoidCapabilityProvider.java:91]: `final var methodName = contextController.getCurrentMethodContext().map(AbstractContext::getName).orElse("na.")`?

**Review by Zsar — APPROVED** [review]

File: src/main/java/eu/tsystems/mms/tic/testerra/plugins/selenoid/webdriver/SelenoidCapabilityProvider.java (hunk 1 above)
Comment (martingrossmann) [reply src/main/java/eu/tsystems/mms/tic/testerra/plugins/selenoid/webdriver/SelenoidCapabilityProvider.java:91]: I asume you like streams? Of course, this is fancier. 😄

Comment (Zsar) [reply src/main/java/eu/tsystems/mms/tic/testerra/plugins/selenoid/webdriver/SelenoidCapabilityProvider.java:91]: No streams here. This is [Optional#map](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Optional.html#map(java.util.function.Function)).

Comment (martingrossmann) [reply src/main/java/eu/tsystems/mms/tic/testerra/plugins/selenoid/webdriver/SelenoidCapabilityProvider.java:91]: Yes... you know what I mean. ;-)

File: src/main/java/eu/tsystems/mms/tic/testerra/plugins/selenoid/webdriver/SelenoidCapabilityProvider.java [hunk 2]
```diff
@@ -45,16 +43,16 @@
...
+public class SelenoidCapabilityProvider implements Consumer<WebDriverRequest>, Loggable, PropertyManagerProvider {
 
     private final boolean VIDEO_ACTIVE = Testerra.Properties.SCREENCASTER_ACTIVE.asBool();
-    private final boolean VNC_ACTIVE = PropertyManager.getBooleanProperty(SelenoidProperties.VNC_ENABLED, SelenoidProperties.Default.VNC_ENABLED);
-    private final String VNC_ADDRESS = PropertyManager.getProperty(SelenoidProperties.VNC_ADDRESS, SelenoidProperties.Default.VNC_ADDRESS);
+    private final boolean VNC_ACTIVE = PROPERTY_MANAGER.getBooleanProperty(SelenoidProperties.VNC_ENABLED, SelenoidProperties.Default.VNC_ENABLED);
+    private final String VNC_ADDRESS = PROPERTY_MANAGER.getProperty(SelenoidProperties.VNC_ADDRESS, SelenoidProperties.Default.VNC_ADDRESS);
 
-    private static final int VIDEO_FRAMERATE = PropertyManager.getIntProperty(SelenoidProperties.VIDEO_FRAMERATE, SelenoidProperties.Default.VIDEO_FRAMERATE);
+    private static final long VIDEO_FRAMERATE = PROPERTY_MANAGER.getLongProperty(SelenoidProperties.VIDEO_FRAMERATE, SelenoidProperties.Default.VIDEO_FRAMERATE);
```
Comment (Zsar) [inline src/main/java/eu/tsystems/mms/tic/testerra/plugins/selenoid/webdriver/SelenoidCapabilityProvider.java:52]: Isn't that... a bit much framerate? I mean, at 60 Hz we are still in single byte territory. What would `9 * 10^19` Hz even look like (other than... an explosion, I guess)?

**Review by Zsar — APPROVED** [review]

File: src/main/java/eu/tsystems/mms/tic/testerra/plugins/selenoid/webdriver/SelenoidCapabilityProvider.java (hunk 2 above)
Comment (martingrossmann) [reply src/main/java/eu/tsystems/mms/tic/testerra/plugins/selenoid/webdriver/SelenoidCapabilityProvider.java:52]: But we do not have a `PROPERTY_MANAGER.getByte()` method.

**Review by mbeuthan — APPROVED** [review]

File: src/main/java/eu/tsystems/mms/tic/testerra/plugins/selenoid/webdriver/SelenoidCapabilityProvider.java (hunk 2 above)
Comment (Zsar) [reply src/main/java/eu/tsystems/mms/tic/testerra/plugins/selenoid/webdriver/SelenoidCapabilityProvider.java:52]: Ah, I see. Well, then. It's all about Thinking Big, right? No love for small values in a big(Int) world.

**Pull request merged**